--- rc80211_minstrel_ht_original.h	2024-06-27 22:18:16.527315277 +0300
+++ rc80211_minstrel_ht.h	2024-06-26 10:38:08.033732700 +0300
@@ -93,6 +93,24 @@
 
 	/* MCS rate group info and statistics */
 	struct minstrel_mcs_group_data groups[MINSTREL_GROUPS_NB];
+
+	//l3s parameters
+	u16 rates[MAX_THR_RATES-1];
+	u16 previous_rate;
+	u16 current_rate;
+	u16 current_state;
+	unsigned int consecutive_successes;
+	unsigned int consecutive_failures;
+	unsigned int consecutive_retries[MAX_THR_RATES-1];
+	unsigned int per[MINSTREL_GROUPS_NB * MCS_GROUP_RATES];
+	unsigned int ack_counter_per_mcs[MINSTREL_GROUPS_NB * MCS_GROUP_RATES];
+	unsigned int total_packets_per_mcs[MINSTREL_GROUPS_NB * MCS_GROUP_RATES];
+	bool recovery;
+	int left_or_right_group;
+	unsigned long probe_interval;
+	unsigned long time_of_probing;
+	bool probe_state;
+	bool first_probe;
 };
 
 struct minstrel_ht_sta_priv {
